File: upwork-intelligence/strip-and-append.mjs
#!/usr/bin/env node
import fs from 'fs';
import path from 'path';

const ROOT = path.dirname(new URL(import.meta.url).pathname);
const out = path.join(ROOT, 'run-results.jsonl');

function stripJob(j) {
  return {
    url: j.url,
    title: j.title,
    published_date: j.published_date,
    created_date: j.created_date,
    job_type: j.job_type,
    budget: j.budget,
    duration: j.duration,
    proposal_count: j.proposal_count,
    experience_level: j.experience_level,
    skills: j.skills,
    client: j.client,
    description_snippet: j.description_snippet,
  };
}

const items = JSON.parse(fs.readFileSync(process.argv[2], 'utf8'));
for (const item of items) {
  const jobs = (item.jobs || item.response?.jobs || []).map(stripJob);
  fs.appendFileSync(out, JSON.stringify({ keyword: item.keyword, group: item.group, jobs }) + '\n');
}
console.log('appended', items.length);

File: upwork-intelligence/_strip_and_append.py
#!/usr/bin/env python3
"""Append one search from MCP JSON file: python3 _strip_and_append.py KEYWORD GROUP response.json"""
import json
import sys
from pathlib import Path

FIELDS = (
    "url", "title", "published_date", "created_date", "job_type", "budget",
    "duration", "proposal_count", "experience_level", "skills", "description_snippet",
)
path = Path(__file__).parent / "search-results-raw.json"
data = json.loads(path.read_text())
resp = json.loads(Path(sys.argv[3]).read_text())
jobs = []
for j in resp.get("jobs") or []:
    jobs.append({k: j.get(k) for k in FIELDS} | {"client": j.get("client")})
entry = {"keyword": sys.argv[1], "group": sys.argv[2], "jobs": jobs}
if resp.get("status") != "ok":
    entry["error"] = True
data["searches"].append(entry)
path.write_text(json.dumps(data))
